graph_weather/data/anemoi_dataloader.py

```diff
@@ -0,0 +1,100 @@
+import numpy as np
+import xarray as xr
+from torch.utils.data import Dataset
+from torchvision.transforms import ToTensor
+
+
+class AnemoiDataset(Dataset):
+    """
+    Dataset for Anemoi weather datasets.
+
+    Args:
+        filepath: Path to the dataset in Zarr format.
+        features: List of features to extract (e.g., "temperature", "geopotential").
+        start_year: Initial year to filter the data. Defaults to 2016.
+        end_year: Ending year to filter the data. Defaults to 2022.
+        resolution: Resolution of the dataset (e.g., "5.625deg").
+        land_sea_mask: Path to the land-sea mask dataset.
+        orography: Path to the orography dataset.
+    """
+
+    def __init__(
+        self,
+        filepath: str,
+        features: list,
+        start_year: int = 2016,
+        end_year: int = 2022,
+        resolution: str = "5.625deg",
+        land_sea_mask: str = None,
+        orography: str = None,
+    ):
+        super().__init__()
+
+        # Check time range validity
+        assert (
+            start_year <= end_year
+        ), f"start_year ({start_year}) cannot be greater than end_year ({end_year})."
+
+        # Load the main dataset
+        self.data = xr.open_zarr(filepath)
+        self.data = self.data.sel(time=slice(str(start_year), str(end_year)))
+
+        # Load additional datasets
+        if land_sea_mask:
+            self.land_sea_mask = xr.open_zarr(land_sea_mask)
+        if orography:
+            self.orography = xr.open_zarr(orography)
+
+        self.features = features
+        self.resolution = resolution
+
+        # Precompute geographical features
+        self._compute_geographical_features()
+
+    def __len__(self):
+        return len(self.data["time"])
+
+    def __getitem__(self, idx):
+        start = self.data.isel(time=idx)
+
+        # Extract features
+        input_data = self._extract_features(start)
+
+        # Add precomputed geographical features
+        input_data = np.concatenate((input_data, self.geo_features), axis=-1)
+
+        # Scale data between 0 and 1
+        input_data = np.clip(input_data, 0, 1)
```

Comment:
I would recommend having this take some normalizer, or mean/stddev to scale between 0 and 1, just clipping the values might throw away information that is important.